plugins/modules/ndo_qos_class_policy.py

```diff
@@ -0,0 +1,463 @@
+        type: int
+        default: 0
+      congestion_algorithm:
+        description:
+        - The congestion algorithm used for this QoS Level.
+        type: str
+        choices: [ tail_drop, wred ]
+        default: tail_drop
+      wred_configuration:
+        description:
+        - The Weighted Random Early Detection (WRED) Algorithm configuration.
+        - This attribute must be specified when O(qos_levels.congestion_algorithm="wred").
+        type: dict
+        suboptions:
+          congestion_notification:
+            description:
+            - The state of Explicit Congestion Notification (ECN) setting.
+            - Enabling Congestion Notification causes the packets that would be dropped to be ECN-marked instead.
+            - The default value is C(disabled).
+            type: str
+            choices: [ enabled, disabled ]
+            default: disabled
+          forward_non_ecn_traffic:
+            description:
+            - Wether to foward Non-ECN Traffic.
+            - This attribute is only used when O(qos_levels.wred_configuration.congestion_notification="enabled").
+            type: bool
+          minimum_threshold:
+            description:
+            - The The minimum queue threshold as a percentage of the maximum queue length for WRED algorithm.
+            - The value must be between 0 and 100.
+            type: int
+            default: 0
+          maximum_threshold:
+            description:
+            - The maximum queue threshold as a percentage of the maximum queue length for WRED algorithm.
+            - The value must be between 0 and 100.
+            - The default value is 100.
+            type: int
+            default: 100
+          probability:
+            description:
+            - The probability value for WRED algorithm.
+            - The probability determines whether the packet is dropped or queued
+              when the average queue size is between the minimum and the maximum threshold values.
+            - The value must be between 0 and 100.
+            type: int
+            default: 0
+          weight:
+            description:
+            - The weight value for WRED algorithm.
+            - Lower weight prioritizes current queue length, while higher weight prioritizes older queue lengths.
+            - The value must be between 0 and 7.
+            type: int
+            default: 0
+      scheduling_algorithm:
+        description:
+        - The Qos Scheduling Algorithm.
+        type: str
+        choices: [ weighted_round_robin, strict_priority ]
+        default: weighted_round_robin
+      bandwidth_allocated:
+        description:
+        - The percentage of total bandwidth allocated to this QoS Level.
+        - The value must be between 0 and 100.
+        type: int
+        default: 20
+      pfc_admin_state:
+        description:
+        - The administrative state of the Priority Flow Control (PFC) policy applied to FCoE traffic.
```

Comment:
```suggestion
        - The administrative state of the Priority Flow Control (PFC) policy.
```
I don't think this applies only to FCoE traffic even if it is mostly used by FCoE

Reply:
changed